crates/edr_napi/src/mock/time.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::Arc;
+
+use edr_generic::GenericChainSpec;
+use edr_napi_core::logger::Logger;
+use edr_solidity::contract_decoder::ContractDecoder;
+use napi::{bindgen_prelude::BigInt, tokio::runtime, Env};
+use napi_derive::napi;
+
+use crate::{
+    cast::TryCast as _,
+    config::{ProviderConfig, TracingConfigWithBuffers},
+    logger::LoggerConfig,
+    provider::Provider,
+    subscription::SubscriptionConfig,
+};
+
+#[napi]
+pub struct MockTime {
+    inner: Arc<edr_provider::time::MockTime>,
+}
+
+#[napi]
+impl MockTime {
+    #[doc = "Creates a new instance of `MockTime` with the current time."]
+    #[napi(factory)]
+    pub fn now() -> Self {
+        Self {
+            inner: Arc::new(edr_provider::time::MockTime::now()),
+        }
+    }
+
+    #[doc = "Adds the specified number of seconds to the current time."]
+    #[napi]
+    pub fn add_seconds(&self, seconds: BigInt) -> napi::Result<()> {
+        let seconds = seconds.try_cast()?;
+
+        self.inner.add_seconds(seconds);
+        Ok(())
+    }
+}
+
+#[doc = "Creates a provider with a mock timer."]
+#[doc = "For testing purposes."]
+#[napi]
+pub fn create_provider_with_mock_timer(
```

Comment:
Would it be possible to define a helper to be used by this and the un-mocked `create_provider`? If not, is there a possibility that this function would get out of sync with that one?

Reply:
After discussing, I decided to revert https://github.com/NomicFoundation/edr/pull/1033/commits/795457ddddc4de21583a15c7f4bb8129d3b005d3

In a follow-up PR I'm making some changes that will allow us to more cleanly share code.